src/intersections/lines.jl

```diff
@@ -9,30 +9,15 @@ The intersection type can be one of three types:
 2. overlap at more than one point
 3. do not overlap nor intersect
 =#
-function intersection(f, l1::Line{3,T}, l2::Line{3,T}) where {T}
+function intersection(f, l1::Line{N,T}, l2::Line{N,T}) where {N,T}
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
 
-  if measure(Tetrahedron(a, b, c, d)) > 0 # not in same plane
-    return @IT NoIntersection nothing f
-  elseif isapprox(norm((b - a) × (c - d)), zero(T), atol=atol(T)^2)
-    if a in l2
-      return @IT OverlappingLines l1 f
-    else # parallel lines
-      return @IT NoIntersection nothing f
-    end
-  else
-    return @IT CrossingLines intersectpoint(l1, l2) f
-  end
-end
-
-function intersection(f, l1::Line{2,T}, l2::Line{2,T}) where {T}
-  a, b = l1(0), l1(1)
-  c, d = l2(0), l2(1)
+  p₁, _, r, rₐ = intersectparameters(a, b, c, d)
 
-  if !isapprox(abs((b - a) × (c - d)), zero(T), atol=atol(T)^2)
-    return @IT CrossingLines intersectpoint(l1, l2) f
-  elseif isapprox(measure(Triangle(a, b, c)), zero(T), atol=atol(T)^2)
+  if r == rₐ == 2
+    return @IT CrossingLines (a + p₁ * (b - a)) f
+  elseif r == rₐ == 1
     return @IT OverlappingLines l1 f
   else
     return @IT NoIntersection nothing f
```

Comment:
looking ahead for the cases of intersecting segments and rays (with each other or any combination with lines) it might be useful to return ranks instead of the fixed IntersectionTypes for lines since other criteria such as the actual value of the parameter will be needed to find the intersectionType.
And there might be more applications to cases where it is important why two lines are not crossing (e.g. parallel vs. skew lines), we have this information still in the ranks but not in the intersection types.